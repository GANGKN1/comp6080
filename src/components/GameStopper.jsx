import React from 'react';
import { Button, Dialog, DialogTitle, DialogContent, Grid, Link } from '@mui/material';
import PropTypes from 'prop-types';
import { Link as RouterLink } from 'react-router-dom';

const GameStopper = ({ show, handleClose, id }) => {
  return (
    <Dialog open={show} onClose={handleClose}>
      <DialogTitle>Game Finished</DialogTitle>
      <DialogContent>
        <h4 className="text-center">Wanna check the results?</h4>
        <Grid container spacing={2}>
          <Grid item md={6} xs={12}>
            <Link component={RouterLink} to={`/results/${id}`} underline="none">
              <Button fullWidth variant="contained" color="primary" id="yes-click">
                Yeah
              </Button>
            </Link>
          </Grid>
          <Grid item md={6} xs={12}>
            <Button
              id="Nah-btn"
              fullWidth
              variant="contained"
              color="error"
              onClick={() => {
                handleClose();
              }}
            >
              Nah
            </Button>
          </Grid>
        </Grid>
      </DialogContent>
    </Dialog>
  );
};
// set the game stopper proptypes
GameStopper.propTypes = {
  setShow: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  allGames: PropTypes.array.isRequired,
  handleShow: PropTypes.func.isRequired,
  setAllGames: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
};

export default GameStopper;
