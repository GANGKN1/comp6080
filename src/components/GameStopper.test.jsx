import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import GameStopper from './GameStopper';
import renderer from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({ adapter: new Adapter() });

const createGameStopperJSON = () => renderer.create(<GameStopper />).toJSON();
const createGameStopperWrapper = () => shallow(<GameStopper />);

describe('GameStopper', () => {
  let step1, step2;

  beforeEach(() => {
    step1 = createGameStopperJSON();
    step2 = createGameStopperWrapper();
  });

  it('The game stopper component should be rendered', () => {
    expect(step2).toHaveLength(1);
    expect(step1).toMatchSnapshot();
  });

  it('Can see the wanna check string', () => {
    const text = step2.find('h4');
    expect(text.text()).toBe('Wanna check the results?');
    expect(step1).toMatchSnapshot();
  });

  it('Can see the button of yeah', () => {
    const text = step2.find('#yes-click');
    expect(text.text()).toBe('Yeah');
    expect(step1).toMatchSnapshot();
  });

  it('Can see the button of Nah', () => {
    const text = step2.find('#Nah-btn');
    expect(text.text()).toBe('Nah');
    expect(step1).toMatchSnapshot();
  });
});
