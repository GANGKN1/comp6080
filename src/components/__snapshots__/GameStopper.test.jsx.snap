// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameStopper  1`] = `null`;

exports[`GameStopper Can check  1`] = `null`;

exports[`GameStopper Can see the  1`] = `null`;

exports[`GameStopper Can see the button of Nah 1`] = `null`;

exports[`GameStopper Can see the button of yeah 1`] = `null`;

exports[`GameStopper Can see the button of yeah 2`] = `null`;

exports[`GameStopper Can see the wanna check string 1`] = `null`;

exports[`GameStopper Can see the wanna check string 2`] = `null`;

exports[`GameStopper The  1`] = `null`;

exports[`GameStopper The game stopper component 1`] = `null`;

exports[`GameStopper The game stopper component should be reder 1`] = `null`;

exports[`GameStopper The game stopper component should be rendered 1`] = `null`;

exports[`GameStopper should have the no button 1`] = `null`;

exports[`GameStopper should have the question text 1`] = `null`;

exports[`GameStopper should have the yes button 1`] = `null`;

exports[`GameStopper should render stop game modal 1`] = `null`;
